models/Duplication_model.slim

```diff
@@ -0,0 +1,167 @@
+initialize() {
+    initializeSLiMOptions(nucleotideBased=T);
+    defineConstant("L", 1010000);              // Genome length
+    defineConstant("DUP_LENGTH", 10000);       // Duplication size
+    defineConstant("DUP_START", L - DUP_LENGTH);  // Start position of duplication
+    defineConstant("N", 500);                  // Population size
+
+    initializeTreeSeq();
+
+    // Generate random nucleotides for the first L - DUP_LENGTH bases
+    randomPart = sample(c("A", "T", "C", "G"), L - DUP_LENGTH - 1, replace=T);
+
+    // Define the segment to duplicate (last DUP_LENGTH bases of randomPart)
+    duplicatedPart = randomPart[(length(randomPart) - DUP_LENGTH):length(randomPart) - 1];
+
+    // Combine randomPart and duplicatedPart to form the full sequence
+    fullSequence = c(randomPart, duplicatedPart);
+
+    // Initialize the ancestral sequence with the full sequence
+    initializeAncestralNucleotides(fullSequence);
+
+
+
+    // Define mutation types
+    initializeMutationTypeNuc("m1", 0.5, "f", 0.0);  // Neutral mutations
+    initializeMutationType("m2", 0.5, "f", 0.0);     // Duplication marker
+    m2.color = "red";
+    m2.convertToSubstitution = F;
+
+    // Define genomic element
+    initializeGenomicElementType("g1", m1, 1.0, mmJukesCantor(1e-7));
+    initializeGenomicElement(g1, 0, L - 1);
+
+    // Set recombination rate
+    initializeRecombinationRate(1e-8);
+}
+
+
+1 early() {
+    defineConstant("simID", getSeed());
+    sim.addSubpop("p1", N);  // Initialize population
+}
+
+recombination() {
+    gm1 = genome1.containsMarkerMutation(m2, DUP_START);
+    gm2 = genome2.containsMarkerMutation(m2, DUP_START);
+    if (!(gm1 | gm2)) {
+        // homozygote non-duplicated
+        // Filter breakpoints that exceed the recombination limit
+        breakpoints = breakpoints[breakpoints < DUP_START];
+
+        // If no breakpoints remain after filtering, return F (no recombination)
+        if (length(breakpoints) == 0) {
+            return F;
+        }
```

Comment:
So, this is not actually what returning F means, for a `recombination()` breakpoint; review section 26.6.  I think you want to just delete these four lines entirely, and just always return T in this branch; you have modified the breakpoints.